common/djangoapps/course_modes/models.py

```diff
@@ -162,23 +166,45 @@ def paid_modes_for_course(cls, course_id):
         return [mode.to_tuple() for mode in found_course_modes]
 
     @classmethod
-    def modes_for_course(cls, course_id):
+    def modes_for_course(cls, course_id, only_selectable=True):
```

Comment:
optional: it's clear enough what `only_selectable` does, but seems like it would require less inline explanation just to call this `exclude_credit_modes` and default it to False (and do the same in the other function).